### PR TITLE
Do not move the test to child job if parallel list number=1

### LINKED ISSUE #5375: No need to trigger a child test job if job can be finished within the TEST_TIME.
Based on the discussion of https://github.com/adoptium/aqa-tests/issues/5368 if a test job can be finished within the TEST_TIME there is no need to trigger a child job (or generate and trigger the child job) to run the test from scratch. It might not be costly for one job however overall for all jdkversions, platforms, categories and impls it should be worth to avoid the triggering the unnecessary child job. 

Besides it makes monitor job easier to less the job layers ( only job triggered by pipeline will be monitored by TRSS)

### PR DESCRIPTION
**Pull request by sophia-guo**
Close #5375

### CODE REVIEW

**Comment by sophia-guo** [comment]
https://ci.adoptium.net/job/Test_openjdk23_hs_special.system_x86-64_linux/10/

**Review by karianna — APPROVED** [review]

File: buildenv/jenkins/JenkinsfileBase [hunk 1]
```diff
@@ -969,7 +973,10 @@ def testBuild() {
 			// prepare environment and compile test projects
 			if ((params.PARALLEL == "NodesByIterations" && NUM_MACHINES > 1)
 				|| (params.PARALLEL == "Dynamic" && (NUM_MACHINES > 1 || (params.TEST_TIME && !params.NUM_MACHINES)))) {
-				setupParallelEnv()
+				if (setupParallelEnv() == 1) {
+					// No need to run tests in parallel
```
[the remark this answers is not in the document]
Comment (sophia-guo) [reply buildenv/jenkins/JenkinsfileBase:977]: Yes, it is echoed https://github.com/adoptium/aqa-tests/pull/5511/files#diff-21a56f751656da4e3e5abee13913f51a269d6eb43235bc16fc4be21ef70deaf2R260

```
16:32:34   Number of test list is 1, no need to run tests in child job.
```

**Review by smlambert — APPROVED** [review]

**Comment by smlambert** [comment]
thanks @sophia-guo - we have already branched for Sept release, could this also be cherrypicked to v1.0.3-release branch please.

File: buildenv/jenkins/JenkinsfileBase [hunk 2]
```diff
@@ -315,7 +319,7 @@ def setupParallelEnv() {
 		if (create_jobs) {
 			parallel create_jobs
 		}
-
+		return NUM_LIST
```
Comment (keithc-ca) [inline buildenv/jenkins/JenkinsfileBase:322]: The variable `NUM_LIST` is not visible here (it's confined to the scope where it's declared, which ends on line 265); our test builds are failing as a result:
```
00:15:35.923  Exception: groovy.lang.MissingPropertyException: No such property: NUM_LIST for class: Script1
```

**Pull request merged**